### DISCUSSION TITLE
Straight forward tutorials on c++ or python with ros2 galactic?

### DISCUSSION THREAD

**Discussion by niebrandon** [iRobotEducation/create3_docs#202] [Questions]
Hi i just bought an irobot create 3 (coming from create 2 running python), and have been using the web python playground, I wanted to try out ros 2, however i just simply dont know where to start. i have looked at the python examples and everything, however when i try to make my project ros2 wont run it. Ive checked the official galactic documentation (not much help) and rclpy is failing to install...if anyone has any simple tutorials or any tips that i can use until the offline sdk comes out it would be greatly appreciated.

**Comment by maddiepero** [comment]
Hi @niebrandon! I'm not sure what type of machine you are working with but we (some students at Tufts University) have been running ros2 very smoothly on VMware or Parallels virtual machines with ubuntu. Here are some instructions we've cooked up about how to install the virtual machine and put ubuntu on it. https://katewujciak.wixsite.com/projectcreate/virtual-machine-instructions 

Then you can install ros2 very easily on the vm https://iroboteducation.github.io/create3_docs/setup/ubuntu2004/ 

Finally, we've made a bunch of starter project ideas and code examples. https://github.com/tuftsceeo/Tufts_Create3_Examples

Let me know if I can provide any clarity on the directions from the website

**Reply by niebrandon** [reply]
Thank you, ive been following these for the past few weeks, and i believe i understand ros a lot better!
